src/views/bootablevolumes/utils/utils.ts

```diff
@@ -81,3 +81,24 @@ export const changeBootableVolumeMetadata =
         ],
       }));
   };
+
+interface InstanceTypesAndSizesObject {
+  [key: string]: string[];
+}
+
+export const getInstanceTypesWithSizes = (
+  instanceTypesNames: string[] = [],
+): InstanceTypesAndSizesObject => {
+  // eslint-disable-next-line prefer-const
+  let instanceTypesAndSizes: InstanceTypesAndSizesObject = {};
+
+  instanceTypesNames.forEach((instanceType) => {
```

Comment:
You could use `reduce` here to create the object. As is, I believe you can use `const` for the object variable since adding properties to the object doesn't change its reference.

Reply:
Yes, after I created this function, I realized I could achieve the same using `reduce`, but this code is much more readable as it is now. Also I haven't found any significant advantage of using reduce, from the performance point of view. I consider readability of this part of the code to be important, because it may not be clear to other programmers what we are trying to achieve in here. We are doing uncommon thing with the specific label - dividing the string into 2 and creating a special object for the data to be easily used in "Default InstanceType" and "Size" dropdowns. Because we are dealing with 2 things but those are somehow connected via one resource (`VirtualMachineClusterInstancetype`) and one label (_instancetype.kubevirt.io/default-instancetype)_, which is IMO not the best idea. But that's how the API for our feature was designed, unfortunately...
Regarding using `const` - I'll try (again), but I think it didn't work when not using `reduce` (not sure why).

Reply:
I'm a huge advocate for readability and I can't see an impact to readability here unless the programmer doesn't understand what `reduce` does, which would be odd for a JavaScript programmer.

In this case, the logic in the body wouldn't really change at all, but it would prevent the requirement for creating an object externally and using side effects from the loop to populate it. 

```
const sizesByInstanceType: SizesByInstanceType = instanceTypeNames.reduce((instanceTypesToSizesMap, instanceType) => {
  const [instanceTypeName, size] = instanceType.split('.');

  instanceTypesToSizesMap[instanceTypeName] !== undefined
    ? instanceTypesToSizesMap[instanceTypeName].push(size)
    : (instanceTypesToSizesMap[instanceTypeName] = [size]);
  });
  
  return instanceTypesToSizesMap;
}, {});
```

Reply:
however I don't quite agree with everything mentioned above and even when the code above won't work, I'll update this part asap ;) 👍🏽  Thanks!